test/framework/cleanup.go

```diff
@@ -53,10 +53,9 @@ func AddCleanupAction(fn func()) CleanupActionHandle {
 func RemoveCleanupAction(p CleanupActionHandle) {
 	cleanupActionsLock.Lock()
 	defer cleanupActionsLock.Unlock()
-	for i, item := range cleanupHookList {
-		if item.actionHandle == p {
+	for i := len(cleanupHookList); i >= 0; i-- {
```

Comment:
Ref https://github.com/gardener/gardener/blob/v1.11.2/test/framework/cleanup.go#L19

Reply:
Reverted. Thanks for spotting it.